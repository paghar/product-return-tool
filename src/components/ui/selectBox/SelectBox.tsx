import React from "react";
import {ISelectBoxItem} from "../../../data/interface";

interface ISelectBox {    
    id?: string,
    className?:string,
    items:ISelectBoxItem[], 
    selectedValue?:string | number,  
    onChange?:(e:any) => void,   
}

const SelectBox = (props:ISelectBox) => {

    const items = props?.items?.map((item:ISelectBoxItem)=>{        
        return <option key={item.id} value={item.id}>{item.text}</option>; 
    });

    return (    
        <select id={props?.id} className={props?.className} value={props.selectedValue} onChange={props.onChange} >                   
            {items}                           
        </select>   
    );
};

export default SelectBox;

