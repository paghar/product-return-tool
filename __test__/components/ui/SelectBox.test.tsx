import renderer from "react-test-renderer";
import SelectBox from "../../../src/components/ui/selectBox/SelectBox";

describe(">>>> SelectBox", () => {
    test("++++ Snapshot SelectBox", () => {
        const component = renderer.create(
            <SelectBox
                id="Radio1"
                items={[]}
                selectedValue=""
                onChange={jest.fn()}                
            />             
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

export {};