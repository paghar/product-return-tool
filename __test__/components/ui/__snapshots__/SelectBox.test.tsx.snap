// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>> SelectBox ++++ Snapshot SelectBox 1`] = `
<select
  id="Radio1"
  onChange={[MockFunction]}
  value=""
/>
`;
